### PR TITLE
[IMP] product: improve pricelist item formview

### PR DESCRIPTION
**Pull request by kp-merucore**
PURPOSE

Improve the current price list line from view for clarity
Current:
https://drive.google.com/file/d/16OL4pEkZtad8nEDgXGvlNeEPvNemVDPO/view?usp=sharing

SPECIFICATIONS

after changes form view is looking accordingly.
- Fixed price
https://drive.google.com/file/d/1hH4gsY_yp60lyXTHtmyV2WKKgkLuAdYq/view?usp=sharing
- Discount
https://drive.google.com/file/d/1texFQGj7ph-6CtQ5YtiEx2eZVkmaPSq8/view?usp=sharing
- Formula
https://drive.google.com/file/d/170et8hNGpK_aNPpL6akmzSKwP6QJFEvU/view?usp=sharing

LINKS

PR https://github.com/odoo/odoo/pull/63733
Task-2373068

### CODE REVIEW

File: addons/product/models/product_pricelist.py [hunk 1]
```diff
@@ -440,6 +440,22 @@ def _default_pricelist_id(self):
...
         'Price', compute='_get_pricelist_item_name_price',
         help="Explicit rule name for this pricelist line.")
+    pricelist_tip = fields.Char(compute='_compute_pricelist_tip')
+
+    @api.depends('compute_price', 'price_discount', 'price_surcharge', 'base', 'price_round')
+    def _compute_pricelist_tip(self):
+        base_selection_vals = {elem[0]: elem[1] for elem in self._fields['base']._description_selection(self.env)}
+        for item in self:
+            price_surcharge = tools.format_amount(item.env, item.price_surcharge, item.currency_id)
+            item.pricelist_tip = '{} with a {}% discount and {} extra fee\nExample: {} * {} + {} = {}'.format(
```
Comment (kba-odoo) [inline addons/product/models/product_pricelist.py:450]: Use the `_()` method to have the tip translated

**Review by kbapt — CHANGES_REQUESTED** [review]

**Comment by mba-odoo** [comment]
@kba-odoo , Done the changes. Thanks

File: addons/product/views/product_pricelist_views.xml [hunk 2]
```diff
@@ -93,77 +93,79 @@
...
+                                    <span>%%</span>
+                                </div>
+                                <field name="price_surcharge" widget="monetary" string="Extra Fee" attrs="{'invisible':[('compute_price', '!=', 'formula')]}"/>
+                                <field name="price_round" string="Rounding Method" attrs="{'invisible':[('compute_price', '!=', 'formula')]}"/>
+                                <label string="Margins" for="price_min_margin" attrs="{'invisible':[('compute_price', '!=', 'formula')]}"/>
+                                <div class="o_row" attrs="{'invisible':[('compute_price', '!=', 'formula')]}">
+                                    <field name="price_min_margin" string="Min. Margin" class="oe_inline" widget="monetary" nolabel="1"/>
+                                    <i class="fa fa-long-arrow-right mx-2 oe_edit_only" aria-label="Arrow icon" title="Arrow"/>
+                                    <field name="price_max_margin" string="Max. Margin" class="oe_inline" widget="monetary" nolabel="1"/>
+                                </div>
```
Comment (Feyensv) [inline addons/product/views/product_pricelist_views.xml:132]: Keep a separate group to display all 'formula' related fields:
```xml
<group name="pricelist_formula" attrs="{'invisible':[('compute_price', '!=', 'formula')]}">
...
```
That way you won't have to specify the same `attrs` for every field inside the group.

**Review by Feyensv — CHANGES_REQUESTED** [review]
As specified in the `price_round` field description: 

> Rounding is applied after the discount and before the surcharge. 
> To have prices that end in 9.99, set rounding 10, surcharge -0.01.

But the tip shown doesn't behave the same way:
image

Please follow the effective price computing logic for the tip or it will be misleading for the users.

**Comment by Feyensv** [comment]
@robodoo r+

**Pull request closed without merging**